x/model/keeper/msg_server_model_version.go

```diff
@@ -109,15 +104,35 @@ func (k msgServer) UpdateModelVersion(goCtx context.Context, msg *types.MsgUpdat
 		return nil, types.NewErrMaxSVLessThanMinSV(modelVersion.MinApplicableSoftwareVersion, msg.MaxApplicableSoftwareVersion)
 	}
 
-	// update existing model version value only if corresponding value in MsgUpdate is not empty
+	if msg.OtaUrl != "" && modelVersion.OtaUrl != "" && msg.OtaUrl != modelVersion.OtaUrl {
+		return nil, types.NewErrOtaURLCannotBeSet(modelVersion.Vid, modelVersion.Pid, modelVersion.SoftwareVersion)
+	}
 
-	// SoftwareVersionValid flag is updated in any case. So pass the existing value to keep it unchanged.
-	modelVersion.SoftwareVersionValid = msg.SoftwareVersionValid
+	if msg.OtaFileSize != 0 && modelVersion.OtaFileSize != 0 && msg.OtaFileSize != modelVersion.OtaFileSize {
+		return nil, types.NewErrOtaFileSizeCannotBeSet(modelVersion.Vid, modelVersion.Pid, modelVersion.SoftwareVersion)
+	}
+
+	if msg.OtaChecksum != "" && modelVersion.OtaChecksum != "" && msg.OtaChecksum != modelVersion.OtaChecksum {
```

Comment:
The check seems wrong. Why do we need it? msg.OtaChecksum should be always mutable without any restrictions.